Classes/System/Solr/SolrConnection.php

```diff
@@ -106,36 +96,46 @@ class SolrConnection
     protected $logger = null;
 
     /**
-     * Adapter class used for the communication
-     *
-     * @var string
+     * @var array
      */
-    protected $adapterClass = Curl::class;
+    protected $clients = [];
 
     /**
-     * This property should be removed at the point Guzzle version 7 is supported by TYPO3
-     *
-     * @var bool|null
-     * @deprecated Will be removed with a future TYPO3 version
+     * @var mixed|ClientInterface
      */
-    protected $psrCompatibilityCheckCache = null;
+    protected $psr7Client;
 
     /**
-     * @var Client[]
+     * @var mixed|RequestFactoryInterface
      */
-    protected $clients = [];
+    protected $requestFactory;
+
+    /**
+     * @var mixed|StreamFactoryInterface
+     */
+    protected $streamFactory;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    protected $eventDispatcher;
 
     /**
      * Constructor
      *
-     * @param Node $readNode,
+     * @param Node $readNode ,
      * @param Node $writeNode
      * @param ?TypoScriptConfiguration $configuration
      * @param ?SynonymParser $synonymParser
      * @param ?StopWordParser $stopWordParser
      * @param ?SchemaParser $schemaParser
      * @param ?SolrLogManager $logManager
      * @param ?EventDispatcherInterface $eventDispatcher
+     * @param ?ClientInterface $psr7Client
+     * @param ?RequestFactoryInterface $requestFactory
+     * @param ?StreamFactoryInterface $streamFactory
```

Comment:
The order of parameters does not match the methods signature.

Reply:
fixed as defined.